### ISSUE TITLE
Error 404/403 for resources from rackcdn.com (HTTPS)

### ISSUE THREAD

**Issue by laukstein** [andrieslouw/imagesweserv#269]
See https://images.weserv.nl/?url=5fbb4b60e5a371522c26-c46478628be7be7c70f96ec65a31d1c7.ssl.cf3.rackcdn.com/Images/ProductImages/Grey_Marl_Zip_Through_Hoodie_124473_1dd1.jpg&w=303&output=webp returns error

```
{"status":"error","code":404,"message":"The requested URL returned error: 403"}
```

Works fine when open directly https://5fbb4b60e5a371522c26-c46478628be7be7c70f96ec65a31d1c7.ssl.cf3.rackcdn.com/Images/ProductImages/Grey_Marl_Zip_Through_Hoodie_124473_1dd1.jpg

Works fine using Cloudinary.

Any fix for images.weserv.nl?

**Comment by laukstein** [comment]
Works fine when defined **https://** in `url` like https://images.weserv.nl/?url=https://5fbb4b60e5a371522c26-c46478628be7be7c70f96ec65a31d1c7.ssl.cf3.rackcdn.com/Images/ProductImages/Grey_Marl_Zip_Through_Hoodie_124473_1dd1.jpg&w=303&output=webp, meaning that images.weserv.nl by the default tries to fetch HTTP and HTTPS, and I think it is wrong to do.

**Comment by andrieslouw** [comment]
Default we fetch over http://, this behavior has been consistent since 2007. Around 2013 we added SSL support using ssl: as prefix. We continued to add support for TLS using this prefix. Unfortunately we cannot change the default behavior easily without breaking things for existing users. We already support https:// to help users.

For more history, please see issue https://github.com/weserv/images/issues/19

I'm sorry, but we do our best to be backwards compatible, and it would be hard for our users to change something we've been doing for 14+ years.

**Comment by laukstein** [comment]
@andrieslouw, if `url` hasn't specified URL protocol shouldn't images.weserv.nl fallback to HTTPS if default HTTP request fails?

**Comment by andrieslouw** [comment]
It maybe could, but then we need to choose what to output if that request also fails. The fail we get here is also quite strict: returning 403 instead of not connecting/not returning anything. 403 more-or-less tells us to not attempt anything further. It is strange that `rackcdn.com` (Akamai) isn't sending any redirect (`Location: ...`) headers, as you would expect for a HTTPS-only host. 

There is also no HSTS-header send (which would be something we'd like to support in the future). `rackcdn.com` is also not on any HSTS-preload list. So no indication of serving over TLS/SSL is being given when doing the original request. There is an `ssl.` part in the subdomain, but that's all...

And we have seen cases of wildly different content being served over HTTPS vs. HTTP, which would lead to confusion if we just try/catch the right protocol.

I'll try to think of a right way to handle this, and will take a look at what browsers are doing nowadays. If we implement try/catch, we also should add a way to disable it.

**Comment by andrieslouw** [comment]
As this is already addressed in #158 , I will mark this as a duplicate.
